salt/states/user.py

```diff
@@ -455,6 +460,11 @@ def present(name,
 
     if gid_from_name:
         gid = __salt__['file.group_to_gid'](name)
+        if not gid:
```

Comment:
This is a logical bug (it will never actually happen in real life, because you always have 'root:root'): you say here if `gid` is `int("0")` and so is `False` in this case, the code meant to bailout, which is incorrect. I would then do so:

```python
if git_from_name:
    gid = __salt__['file.group_to_gid'](name)
    if gid == '':
        gid = None
    # ^
    # |
    # +--- at this point you've finished with setting
    #      a proper type of "gid" once and for all here.
    #
    # Now bailout if things are wrong with the default groups:
    if gid is None:
        ret['comment'] = 'be be be....'  # Please keep this in one line
        ...
        return ret
```

In this case function `_changes` (called twice in this scope) will get your `gid=None` instead of `gid=''` and so the issue is fine. But you have no scattered checks all around the place with a semi-bug introduced.